oasis-reference/src/main/kotlin/com/empowerops/volition/ref_oasis/OptimizerController.kt

```diff
@@ -0,0 +1,230 @@
+
+        nameColumn.cellValueFactory = TreeItemPropertyValueFactory<Parameter, String>("name")
+        valueColumn.cellValueFactory = TreeItemPropertyValueFactory<Parameter, String>("value")
+        lbColumn.cellValueFactory = TreeItemPropertyValueFactory<Parameter, String>("lb")
+        upColumn.cellValueFactory = TreeItemPropertyValueFactory<Parameter, String>("ub")
+
+        nodesList.selectionModel.selectedItemProperty().addListener { src, oldV, newV -> showNode(newV) }
+
+        timeOutTextField.setOnKeyPressed{ event ->
+            if(event.code == KeyCode.ESCAPE){
+                //discard
+                timeOutTextField.text = endpoint.simulationsByName.getValue(nodesList.selectedItem!!).timeOut!!.toMillis().toString()
+                view.requestFocus()
+            }
+            else if (event.code == KeyCode.ENTER) {
+                //commit
+                try{
+                    endpoint.setDuration(nodesList.selectedItem, Duration.ofMillis(timeOutTextField.text.toLong()))
+                }
+                catch (e : NumberFormatException){
+                    timeOutTextField.text = ""
+                }
+                view.requestFocus()
+            }
+        }
+
+        useTimeout.selectedProperty().addListener{s, oldV, newV ->
+            if(newV){
+                endpoint.setDuration(nodesList.selectedItem, Duration.ZERO)
+                timeOutTextField.text = "0"
+            }
+            else{
+                endpoint.setDuration(nodesList.selectedItem, null)
+                timeOutTextField.text = ""
+            }
+        }
+
+        timeOutTextField.disableProperty().bind(useTimeout.selectedProperty().not())
+    }
+
+    private fun showNode(newV: String?) {
+        if(newV == null){
+            selectedNodeInfoBox.isDisable = true
+        }
+        else{
+            selectedNodeInfoBox.isDisable = false
+            val sim = endpoint.simulationsByName.getValue(newV)
+            descriptionLabel.text = newV
+            val buildTree = buildTree(sim)
+            paramTreeView.root = buildTree
+            paramTreeView.isShowRoot = false
+            statusLabel.text = sim.description
+            if(sim.timeOut!= null){
+                useTimeout.isSelected = true
+                timeOutTextField.text = sim.timeOut.toMillis().toString()
+            }
+            else{
+                useTimeout.isSelected = false
+                timeOutTextField.text = ""
+            }
+        }
+    }
+
+    fun setData(viewData: ViewData, optimizerEndpoint: OptimizerEndpoint){
+        nodesList.items = viewData.nodes
+        messageTableView.items = viewData.allMessages
+        resultTableView.items = viewData.resultList
+        endpoint = optimizerEndpoint
+        optimizerStatusLabel.textProperty().bind(viewData.currentEvaluationStatus)
+        viewData.updateList.addListener { o: Observable ->
+            if(viewData.updateList.contains(nodesList.selectedItem)){
```

Comment:
`if(nodesList.selectedItem in viewData.updateList)`?

Reply:
this is removed, but I will do more of that